### ISSUE TITLE
Update errors

### ISSUE THREAD

**Issue by boozook** [aerhard/linter-autocomplete-jing#27]
``` log
Updating to “linter-jing@0.5.3” failed.
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open '/Users/ak/.atom/packages/linter-autocomplete-jing/node_modules/os-tmpdir/index.js'
    at Error (native)
```

``` log
Updating to “linter-autocomplete-jing@0.5.3” failed.
ENOTEMPTY: directory not empty, rmdir '/Users/ak/.atom/packages/linter-autocomplete-jing/node_modules/atom-linter'
```

``` log
[linter-autocomplete-jing] Unexpected server start message "Error: Unable to access jarfile /Users/ak/.atom/packages/linter-autocomplete-jing/vendor/xml-tools-server-0.4.2.jar "
Error
    at Error.ServerProcess.Error (/Users/ak/.atom/packages/linter-autocomplete-jing/lib/serverProcess.js:162:34)
    at Socket.onData (/Users/ak/.atom/packages/linter-autocomplete-jing/lib/serverProcess.js:62:16)
    at emitOne (events.js:90:13)
    at Socket.emit (events.js:182:7)
    at readableAddChunk (_stream_readable.js:153:18)
    at Socket.Readable.push (_stream_readable.js:111:10)
    at Pipe.onread (net.js:531:20)
```

![screen shot 2016-10-06 at 16 50 42](https://cloud.githubusercontent.com/assets/888526/19155025/f2583226-8bed-11e6-807f-976412891e3d.png)

**Comment by aerhard** [comment]
Hi - thank you for reporting that error. I'll try to reproduce it on my computer. The error messages seem to indicate that the package got updated only partially -- the JavaScript code looks for 'xml-tools-server-0.4.2.jar' while the name of that jar file deployed with v0.5.3 should be 'xml-tools-server-0.4.4.jar'. Have you already tried uninstalling the package and then re-installing it again (shut down Atom and run `apm uninstall linter-autocomplete-jing` and then `apm install linter-autocomplete-jing`)?

**Comment by boozook** [comment]
Now I did uninstall jing (both two O_o), then restart Atom.app and get the next error:

``` log
Error: Cannot find module 'atom-package-deps'
    at Module._resolveFilename (module.js:339:15)
    at Function.Module._resolveFilename (/Applications/Atom.app/Contents/Resources/app.asar/src/module-cache.js:383:52)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (/Applications/Atom.app/Contents/Resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.activate (/Users/ak/.atom/packages/linter-autocomplete-jing/lib/main.js:76:5)
    at Package.module.exports.Package.activateNow (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:183:20)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:156:32
    at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:92:15)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:149:26
    at Package.module.exports.Package.activate (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:146:34)
    at PackageManager.module.exports.PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:538:21)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:519:29
    at Config.module.exports.Config.transactAsync (/Applications/Atom.app/Contents/Resources/app.asar/src/config.js:337:18)
    at PackageManager.module.exports.PackageManager.activatePackages (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:514:19)
    at PackageManager.module.exports.PackageManager.activate (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:497:46)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/atom-environment.js:765:28
```

![screen shot 2016-10-07 at 15 43 23](https://cloud.githubusercontent.com/assets/888526/19190265/9e42fb4c-8cad-11e6-9cb8-9385f1a6f7e5.png)

---

**UPDATE:**

It's very funny! Packages wasn't removed.
![screen shot 2016-10-07 at 15 46 56](https://cloud.githubusercontent.com/assets/888526/19190297/c689a876-8cad-11e6-8888-480ee2df9a4c.png)

**Comment by aerhard** [comment]
oh -- strange that it didn't remove the old package. Something went terribly wrong with the update. I've renamed the package after adding autocomplete functionality, sorry if that caused this issue (I thought renaming was a 'safe' operation). In any case, `apm uninstall ...` didn't seem to work so I'd recommend to delete the package directories manually. According to your screenshots Atom's packages are stored in `/Users/ak/.atom/packages`, so after closing Atom you should be able to remove the directories by executing 

```
rm -rf /Users/ak/.atom/packages/linter-autocomplete-jing
rm -rf /Users/ak/.atom/packages/linter-jing
```

When you open Atom after these directories are gone, no error messages should appear and you should be able to do a fresh install of the `linter-autocomplete-jing` package from within Atom. I'm still wondering why the update did fail in the first place -- couldn't reproduce the behavior on my computer but I've only tried with the most recent version of Atom. I'll check if it can be reproduced with an older one

**Comment by aerhard** [comment]
Could reproduce the error message `Unexpected server start message "Error: Unable to access jarfile /Users/ak/.atom/packages/linter-autocomplete-jing/vendor/xml-tools-server-0.4.2.jar "` with the latest Atom version. The cause for this is that updating the package doesn't reload the JavaScript code. This doesn't seem to be an issue specific to the linter-autocomplete-jing package since I get the same behavior with a test package that has no dependencies and only logs the current version when it gets activated. After that test package gets updated the activation code is run which still logs the old version number. For now I'm going to look for a way to deal with this in a more elegant way (prevent the error message, give a hint that using the new version requires reloading the current Atom window with Ctrl-Alt-R, ideally make it possible to continue editing in the current session with the old version)

**Comment by aerhard** [comment]
I've now published v0.6.0 which enables hot reloading the package code after updates. Hot reloading will only be available when you update FROM v0.6.0 or above. Nevertheless, an update from v0.5.3 to v0.6.0 should allow you to continue working in Atom immediately without showing an error popup. Please add a comment if there are any issues
